utils/logalloc.cc

```diff
@@ -1117,20 +1117,20 @@ static segment_pool& get_shard_segment_pool() noexcept {
 
 static thread_local segment_pool& shard_segment_pool = get_shard_segment_pool();
 
-void segment::record_alloc(segment::size_type size) {
+void segment::record_alloc(segment::size_type size) noexcept {
```

Comment:
Some can, but not all of them due to keeping the header thin and independent as possible (using forward declaration rather than including full class definitions for  dependencies)